File: apps/sepuling/src/main/java/smecalculus/bezmen/messaging/SepulkaClientImpl.java
package smecalculus.bezmen.messaging;

import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import smecalculus.bezmen.core.SepulkaService;
import smecalculus.bezmen.validation.EdgeValidator;

@RequiredArgsConstructor
public class SepulkaClientImpl implements SepulkaClient {

    @NonNull
    private EdgeValidator validator;

    @NonNull
    private SepulkaMsgMapper mapper;

    @NonNull
    private SepulkaService service;

    @Override
    public SepulkaNewResponseEdge register(SepulkaNewRequestEdge requestEdge) {
        validator.validate(requestEdge);
        var request = mapper.toDomain(requestEdge);
        var response = service.register(request);
        return mapper.toEdge(response);
    }
}

File: apps/sepuling-kotlin/src/main/kotlin/smecalculus/bezmen/messaging/SepulkaClientImpl.kt
package smecalculus.bezmen.messaging

import smecalculus.bezmen.core.SepulkaService
import smecalculus.bezmen.validation.EdgeValidator

class SepulkaClientImpl(
    private val validator: EdgeValidator,
    private val mapper: SepulkaMessageMapper,
    private val service: SepulkaService
) : SepulkaClient {

    override fun register(requestEdge: MessageEm.RegistrationRequest): MessageEm.RegistrationResponse {
        validator.validate(requestEdge)
        val request = mapper.toDomain(requestEdge)
        val response = service.register(request)
        return mapper.toEdge(response)
    }
}
